cache/backend/cloud_storage.go

```diff
@@ -0,0 +1,60 @@
+package backend
+
+import (
+	"context"
+	"io"
+
+	"github.com/meltwater/drone-cache/cache"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+)
+
+// cloudStorageBackend is an Cloud Storage implementation of the Backend
+type cloudStorageBackend struct {
+	bucket     string
+	acl        string
+	encryption string
+	client     *storage.Client
+}
+
+// newCloudStorage returns a new Cloud Storage remote Backend implemented
+func newCloudStorage(bucket, acl, encryption string, opts ...option.ClientOption) (cache.Backend, error) {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return &cloudStorageBackend{
+		bucket:     bucket,
+		acl:        acl,
+		encryption: encryption,
+		client:     client,
+	}, nil
+}
+
+// Get returns an io.Reader for reading the contents of the file
+func (c *cloudStorageBackend) Get(p string) (io.ReadCloser, error) {
+	bkt := c.client.Bucket(c.bucket)
+	obj := bkt.Object(p)
+	if c.encryption != "" {
+		obj = obj.Key([]byte(c.encryption))
+	}
+
+	// TODO: use a timeout or cancel context
+	return obj.NewReader(context.TODO())
```

Comment:
Let's use `context.Background()` for now. We should introduce a `timeout` value for operations. I'll open an issue for it.